src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs

```diff
@@ -289,16 +294,19 @@ public void AddToPending(ChainedHeaderBlock chainedHeaderBlock)
         {
             lock (this.blocksCacheLock)
             {
-                bool added = this.pendingBlocksCache.TryAdd(chainedHeaderBlock.ChainedHeader.HashBlock, chainedHeaderBlock);
-
-                if (added)
+                ChainedHeaderBlock existing = null;
+                this.pendingBlocksCache.TryGetValue(chainedHeaderBlock.ChainedHeader.HashBlock, out existing);
```

Comment:
Why replace one operation with two?

Most of the time `TryAdd` will work so I would suggest just leave it like that.